hypothesis-python/src/hypothesis/strategies/_internal/datetime.py

```diff
@@ -314,3 +318,131 @@ def timedeltas(
     if min_value == max_value:
         return just(min_value)
     return TimedeltaStrategy(min_value=min_value, max_value=max_value)
+
+
+@defines_strategy(force_reusable_values=True)
+def timezone_keys(
+    *,
+    # allow_alias: bool = True,
+    # allow_deprecated: bool = True,
+    allow_prefix: bool = True,
+) -> SearchStrategy[str]:
+    """A strategy for :wikipedia:`IANA timezone names <List_of_tz_database_time_zones>`.
+
+    As well as timezone names like ``"UTC"``, ``"Australia/Sydney"``, or
+    ``"America/New_York"``, this strategy can generate:
+
+    - Aliases such as ``"Antarctica/McMurdo"``, which links to ``"Pacific/Auckland"``.
+    - Deprecated names such as ``"Antarctica/South_Pole"``, which *also* links to
+      ``"Pacific/Auckland"``.  Note that most but
+      not all deprecated timezone names are also aliases.
+    - Timezone names with the ``"posix/"`` or ``"right/"`` prefixes, unless
+      ``allow_prefix=False``.
+
+    These strings are provided separately from Tzinfo objects - such as ZoneInfo
+    instances from the timezones() strategy - to facilitate testing of timezone
+    logic without needing workarounds to access non-canonical names.
+
+    .. note::
+
+        The :mod:`python:zoneinfo` module is new in Python 3.9, so you will need
+        to install the :pypi:`backports.zoneinfo` module on earlier versions, and
+        the :pypi:`importlib_resources` backport on Python 3.6.
+
+        ``pip install hypothesis[zoneinfo]`` will install these conditional
+        dependencies if and only if they are needed.
+
+    On Windows, you may need to access IANA timezone data via the :pypi:`tzdata`
+    package.  For non-IANA timezones, such as Windows-native names or GNU TZ
+    strings, we recommend using :func:`~hypothesis.strategies.sampled_from` with
+    the :pypi:`dateutil` package, e.g. :meth:`dateutil:dateutil.tz.tzwin.list`.
+    """
+    # check_type(bool, allow_alias, "allow_alias")
+    # check_type(bool, allow_deprecated, "allow_deprecated")
+    check_type(bool, allow_prefix, "allow_prefix")
+    if zoneinfo is None:  # pragma: no cover
+        raise ModuleNotFoundError(
+            "The zoneinfo module is required, but could not be imported.  "
+            "Run `pip install hypothesis[zoneinfo]` and try again."
+        )
+
+    available_timezones = ("UTC",) + tuple(sorted(zoneinfo.available_timezones()))
+
+    # TODO: filter out alias and deprecated names if disallowed
+
+    # When prefixes are allowed, we first choose a key and then flatmap to get our
+    # choice with one of the available prefixes.  That in turn means that we need
+    # some logic to determine which prefixes are available for a given key:
+    try:
+        from importlib import resources
+    except ImportError:
+        try:
+            import importlib_resources as resources  # type: ignore
+        except ImportError as err:
+            raise type(err)(
```

Comment:
I don't *think* so, because the other case is specific to the data-only `tzdata` package - whereas attempting to import this backport can raise other import errors too.